### PR TITLE
Update AQT Backend

### PR DESCRIPTION
**Pull request by jbrixon**

### CODE REVIEW

**Comment by jbrixon** [comment]
@fg-aqt can you check the docs please? As discussed there are a few mentions of signing up via the Gateway Portal which are no longer relevant. Cheers!

File: cirq-aqt/cirq_aqt/aqt_sampler.py [hunk 1]
```diff
@@ -99,12 +104,64 @@ def _generate_json(
...
+
+        for legacy_op in json.loads(json_str):
+            if number_of_measurements > 0:
+                raise ValueError(
+                    "Need exactly one `MEASURE` operation at the end of the circuit."
+                )
+
+            instruction = {}
+
+            match legacy_op[0]:
```
Comment (wilfried-huss) [inline cirq-aqt/cirq_aqt/aqt_sampler.py]: Probably Cirq needs to support Python versions < 3.10

See: https://github.com/quantumlib/Cirq/blob/main/setup.py#L67C5-L67C31

**Comment by fg-aqt** [comment]
> @fg-aqt can you check the docs please? As discussed there are a few mentions of signing up via the Gateway Portal which are no longer relevant. Cheers!

I updated the access.md and pushed the changes. I noticed that the functionality to retrieve the available workspaces with the respective resources is missing. Can you please add this?

I also did some preliminary updates of the getting_started instructions, we need to work some more on these.

File: cirq-aqt/cirq_aqt/aqt_sampler.py [hunk 2]
```diff
@@ -26,9 +26,10 @@
 import time
 import uuid
 from typing import cast, Dict, List, Sequence, Tuple, Union
+from urllib.parse import urljoin
 
 import numpy as np
-from requests import put
+from requests import put, post, get
```
Comment (airwoodix) [inline cirq-aqt/cirq_aqt/aqt_sampler.py]:
```suggestion
from requests import post, get
```

`put` is not used anymore

File: cirq-aqt/cirq_aqt/aqt_sampler.py [hunk 3]
```diff
@@ -120,7 +176,7 @@ def _send_json(
```
Comment (airwoodix) [inline cirq-aqt/cirq_aqt/aqt_sampler.py:223]: The docstring needs an update (references PUT requests).

File: cirq-aqt/cirq_aqt/aqt_sampler_test.py [hunk 4]
```diff
@@ -39,74 +43,116 @@ def json(self):
...
+        with (
+            mock.patch('cirq_aqt.aqt_sampler.post', return_value=e_return),
+            mock.patch('cirq_aqt.aqt_sampler.get', return_value=GetResultReturn()),
+        ):
             theta = sympy.Symbol('theta')
             num_points = 1
             max_angle = np.pi
             repetitions = 10
-            sampler = AQTSampler(remote_host="http://localhost:5000", access_token='testkey')
+            sampler = AQTSampler(remote_host="http://localhost:7777/api/v1/", access_token='testkey', workspace="default", resource="test")
```
Comment (airwoodix) [inline cirq-aqt/cirq_aqt/aqt_sampler_test.py]: Is the `remote_host` here indeed arbitrary? Should this be obvious when reading the code?

**Review by airwoodix — APPROVED** [review]

File: cirq-aqt/cirq_aqt/aqt_sampler.py (hunk 2 above)
Comment (jbrixon) [reply cirq-aqt/cirq_aqt/aqt_sampler.py]: Thanks! Just caught this too, fix is in :-)

File: cirq-aqt/cirq_aqt/aqt_sampler.py (hunk 3 above)
Comment (jbrixon) [reply cirq-aqt/cirq_aqt/aqt_sampler.py:223]: Just caught this too, thanks!

File: cirq-aqt/cirq_aqt/aqt_sampler_test.py (hunk 4 above)
Comment (jbrixon) [reply cirq-aqt/cirq_aqt/aqt_sampler_test.py]: They are in three places, yes! I obviously tested against a local Arnica instance as I changed the URL, but they are arbitrary with the patches applied. I removed where possible to make it clearer that it's irrelevant. Cheers!

**Comment by jbrixon** [comment]
> I updated the access.md and pushed the changes. I noticed that the functionality to retrieve the available workspaces with the respective resources is missing. Can you please add this?

Is in with latest commit

File: cirq-aqt/cirq_aqt/aqt_sampler.py [hunk 5]
```diff
@@ -238,15 +358,15 @@ def _send_json(
         self,
         *,
         json_str: str,
-        id_str: Union[str, uuid.UUID],
+        id_str: str,
```
Comment (fg-aqt) [inline cirq-aqt/cirq_aqt/aqt_sampler.py:361]: We could use this to distinguish between a noisy and ideal simulator, similarly to the way we do it in Qiskit with the offline simulators. If we do it, we could also add the offline simulators to the `fetch_resources` function.

**Review by fg-aqt — APPROVED** [review]

File: cirq-aqt/cirq_aqt/aqt_sampler.py (hunk 5 above)
Comment (jbrixon) [reply cirq-aqt/cirq_aqt/aqt_sampler.py:361]: As discussed offline, this was always determined by the simulate_ideal argument and was left like that. The id_str arg is used on the regular sampler to define the label for the circuit, so such a change in function might be counter-intuitive.

File: cirq-aqt/cirq_aqt/aqt_device.py [hunk 6]
```diff
@@ -122,12 +123,15 @@ def get_crosstalk_operation(
             xtlk_arr[idx] = 0
         xtlk_op_list = []
         op_str = get_op_string(operation)
+        # if op_str == "R":
+        #     gate = cast(cirq.PhasedXPowGate, gate_dict[op_str])
+        # else:
```
Comment (fg-aqt) [inline cirq-aqt/cirq_aqt/aqt_device.py]: This can be removed

Comment (fg-aqt) [reply cirq-aqt/cirq_aqt/aqt_device.py]: So potentially can the introduced empty lines

**Review by fg-aqt — CHANGES_REQUESTED** [review]

File: cirq-aqt/cirq_aqt/aqt_device.py (hunk 6 above)
Comment (jbrixon) [reply cirq-aqt/cirq_aqt/aqt_device.py]: The whole block is useless so have removed. Cheers for the catch.

**Review by fg-aqt — APPROVED** [review]

**Pull request merged**